launcher/src/main/java/org/wildfly/core/launcher/CliCommandBuilder.java

```diff
@@ -0,0 +1,654 @@
+     * Creates a command builder for a CLI instance.
+     *
+     * @param wildflyHome the path to the WildFly home directory
+     *
+     * @return a new builder
+     */
+    public static CliCommandBuilder of(final String wildflyHome) {
+        return new CliCommandBuilder(new Environment(wildflyHome));
+    }
+
+    /**
+     * Sets the hostname and port to connect to.
+     * <p>
+     * This sets both the {@code --connect} and {@code --controller} arguments.
+     * </p>
+     *
+     * @param controller the controller argument to use
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setConnection(final String controller) {
+        addCliArgument(CliArgument.CONNECT);
+        addCliArgument(CliArgument.CONTROLLER, controller);
+        return this;
+    }
+
+    /**
+     * Sets the hostname and port to connect to.
+     * <p>
+     * This sets both the {@code --connect} and {@code --controller} arguments.
+     * </p>
+     *
+     * @param hostname the host name
+     * @param port     the port
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setConnection(final String hostname, final int port) {
+        addCliArgument(CliArgument.CONNECT);
+        setController(hostname, port);
+        return this;
+    }
+
+    /**
+     * Sets the protocol, hostname and port to connect to.
+     * <p>
+     * This sets both the {@code --connect} and {@code --controller} arguments.
+     * </p>
+     *
+     * @param protocol the protocol to use
+     * @param hostname the host name
+     * @param port     the port
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setConnection(final String protocol, final String hostname, final int port) {
+        addCliArgument(CliArgument.CONNECT);
+        setController(protocol, hostname, port);
+        return this;
+    }
+
+    /**
+     * Sets the hostname and port to connect to.
+     *
+     * @param controller the controller argument to use
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setController(final String controller) {
+        addCliArgument(CliArgument.CONTROLLER, controller);
+        return this;
+    }
+
+    /**
+     * Sets the hostname and port to connect to.
+     *
+     * @param hostname the host name
+     * @param port     the port
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setController(final String hostname, final int port) {
+        setController(String.format("%s:%d", hostname, port));
+        return this;
+    }
+
+    /**
+     * Sets the protocol, hostname and port to connect to.
+     *
+     * @param protocol the protocol to use
+     * @param hostname the host name
+     * @param port     the port
+     *
+     * @return the builder
+     */
+    public CliCommandBuilder setController(final String protocol, final String hostname, final int port) {
+        setController(String.format("%s://%s:%d", protocol, hostname, port));
```

Comment:
Do we want to fully support parsing it like the `org.jboss.as.network.NetworkUtils` does?